### DISCUSSION TITLE
[rejected] Efektywne pobieranie wystawionych ocen

### DISCUSSION THREAD

**Discussion by javaguru-pl** [allegro/allegro-api#4814] [Suggestions - rejected]
Witam,

Obecnie do pobierania **ocen** wystawionych od kupujących - jest dostępna metoda:
**[https://developer.allegro.pl/documentation/#operation/offerRatingGET](https://developer.allegro.pl/documentation/#operation/offerRatingGET)**
Problem w tym że wymaga ona pojedynczego numeru aukcji jako parametr. Czy była by możliwość dodania metody w API pobierającej **wszystkie oceny wystawione dla sprzedawcy**? (np. podając jako argument zakres dat, lub ID ostatnio pobranej oceny - podobnie jak w dzienniku zdarzeń).

Niestety obecny sposób pobierania ocen jest **NIEMOŻLIWY do realizacji** w przypadku aplikacji, która obsługuje tysiące kont Allegro, a każde z nich ma tysiące aukcji na stanie (tzn. technicznie niewykonalne jest dzienne wywołanie tej metody dziesiątki milionów razy dla każdej aukcji z osobna).
Z góry dziękuję za odpowiedź.

**Comment by Lukasz-Zurek** [comment]
@javaguru-pl Zasób, o którym wspominasz: [GET /sale/offers/{offerId}/rating](https://developer.allegro.pl/documentation/#operation/offerRatingGET) służy do pobierania ocen produktów przypisanych do Twoich ofert.

Jeśli natomiast chodzi o możliwość pobrania ocen (dotyczących transakcji) wystawionych od kupujących, służy do tego: [GET /sale/user-ratings](https://developer.allegro.pl/documentation/#operation/getUserRatingsUsingGET). W odpowiedzi otrzymujesz oceny sprzedaży użytkownika. Pamiętaj, że zasób ten wymaga "bearer-token-for-user", czyli działania w kontekście zalogowanego użytkownika.

Więcej informacji znajdziesz w [naszym poradniku](https://developer.allegro.pl/account/#2-ocena-sprzeda%C5%BCy).

**Reply by javaguru-pl** [reply]
Witam Panie Łukaszu,

Bardzo dziękuję za podpowiedź;)
Nie zauważyłem tej metody, gdyż szukałem w sekcji "Offer rating", a metoda /sale/user-ratings jest w sekcji "Information about user". To mnie zmyliło;)
